### DISCUSSION TITLE
Does Raytha have a story for multitenancy?

### DISCUSSION THREAD

**Discussion by Jogai** [RaythaHQ/raytha#124] [General]
Just a quick question. Maybe it should be on the ideas list, maybe its there already. But I literally just found this just before signing off for the weekend.

**Comment by apexdodge** [comment]
Thanks for your comment. This is under consideration, the idea of some kind of "multi site" capability.

Multi-tenant can be implemented in primarily two ways.

- 1 web app, single database, but data is segmented by row key of customer/site.
- 1 web app, multiple databases. We share compute, but 1 database per customer/site.

Each with advantage / disadvantages.

However, with the advent and rise of docker / kubernetes, the above scenario is not as essential as it used to be. Spinning up a container + db into a kubernetes / azure sql elastic pool and then use nginx or service mesh to set routing has a lot of advantages too.

We intend to build a cloud version of the raytha platform that will use the above description. Depending on how that works out, we will re-evaluate.

**Comment by Jogai** [comment]
You're right about the docker / kubernetes part. Although there could be some value in an overarching application, because its nice to be able to provision users, toggle features, or adjust configs per tenant
